src/xref2/ident.ml

```diff
@@ -199,21 +208,29 @@ end
 
 module Name = struct
   let rec signature : signature -> string = function
-    | `LRoot (n, _) -> UnitName.to_string n
+    | `LRoot (n, _) -> ModuleName.to_string n
     | `LModule (n, _) -> ModuleName.to_string n
     | `LResult (x, _) -> signature x
     | `LParameter (n, _) -> ParameterName.to_string n
     | `LModuleType (n, _) -> ModuleTypeName.to_string n
 
+  let module_' : module_ -> ModuleName.t = function
```

Comment:
I prefer the original `typed_module` than `module'` - same for `functor_parameter'` below.